packages/core/src/utils.ts

```diff
@@ -709,34 +713,51 @@ export const assertValidUpgrade = async (
     parsedConfig.contracts
   )) {
     if ((await provider.getCode(contractConfig.proxy)) !== '0x') {
-      const proxyAdmin = await getEIP1967ProxyAdminAddress(
-        provider,
-        contractConfig.proxy
-      )
+      isUpgrade = true
 
-      let managerIsValidUpgrader = true
-      const signer = new ethers.VoidSigner(chugSplashManagerAddress, provider)
-      const contract = new ethers.Contract(
-        contractConfig.proxy,
-        ProxyABI,
-        signer
-      )
-      try {
-        await contract.callStatic.upgradeTo(
-          '0x1111111111111111111111111111111111111111'
+      if (contractConfig.proxyType === 'oz-uups') {
+        // We must manually check that the ChugSplashManager can call the UUPS proxy's `upgradeTo`
+        // function because OpenZeppelin UUPS proxies can implement arbitrary access control
+        // mechanisms.
+        const chugsplashManager = new ethers.VoidSigner(
+          chugSplashManagerAddress,
+          provider
+        )
+        const UUPSProxy = new ethers.Contract(
+          contractConfig.proxy,
+          ProxyABI,
+          chugsplashManager
+        )
+        try {
+          // Attempt to staticcall the `upgradeTo` function on the proxy from the ChugSplashManager's
+          // address. Note that we upgrade to the address of the ChugSplashRegistry's proxy to ensure
+          // that the call doesn't fail due to this check:
+          // https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/ERC1967/ERC1967Upgrade.sol#L44
+          await UUPSProxy.callStatic.upgradeTo(
+            CHUGSPLASH_REGISTRY_PROXY_ADDRESS
```

Comment:
This doesn't work b/c the OpenZeppelin does not consider the interface of the ChugSplashRegistry to be UUPS compatible. So if you run the e2e tests, the UUPS test will fail: 
`yarn test:hardhat`

If you print the error you'll see it's message is: 
`ERC1967Upgrade: new implementation is not UUPS`

This error occurs b/c the target does not include the `proxiableUUID()` function: 
https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/dd8ca8adc47624c5c5e2f4d412f5f421951dcc25/contracts/proxy/ERC1967/ERC1967UpgradeUpgradeable.sol#L91

To be fair, this issue also existed in my implementation, your refactor just uncovered it. The solution is to use an address here that conforms to the full UUPS expected spec. So we'd either need to add the `proxiableUUID()` function to the ChugSplashRegistry or use the address of a different contract that implements it.